web/app/controllers/api/v1/User.java

```diff
@@ -61,30 +71,39 @@ public static Result updateSettings()
         return ok(result);
     }
 
-    public static Result getAllCompanyUsers()
-    {
+    public static Result getAllCompanyUsers() {
+        List<CompanyUser> users = (List<CompanyUser>) Cache.get(CACHE_INTERNAL_USERS);
+        if (users == null || users.size() == 0) {
+            users = UserDAO.getAllCompanyUsers();
+            Cache.set(CACHE_INTERNAL_USERS, users, 3600); // cache for an hour
+        }
         ObjectNode result = Json.newObject();
-
         result.put("status", "ok");
-        result.set("employees", Json.toJson(UserDAO.getAllCompanyUsers()));
+        result.set("employees", Json.toJson(users));
         return ok(result);
     }
 
-    public static Result getAllGroups()
-    {
+    public static Result getAllGroups() {
+        List<Group> groups = (List<Group>) Cache.get(CACHE_INTERNAL_GROUPS);
+        if (groups == null || groups.size() == 0) {
+            groups = UserDAO.getAllGroups();
+            Cache.set(CACHE_INTERNAL_GROUPS, groups, 3600); // cache for an hour
+        }
         ObjectNode result = Json.newObject();
-
         result.put("status", "ok");
-        result.set("groups", Json.toJson(UserDAO.getAllGroups()));
+        result.set("groups", Json.toJson(groups));
         return ok(result);
     }
 
-    public static Result getAllUserEntities()
-    {
+    public static Result getAllUserEntities() {
+        List<UserEntity> entities = (List<UserEntity>) Cache.get(CACHE_INTERNAL_ENTITIES);
+        if (entities == null || entities.size() == 0) {
+            entities = UserDAO.getAllUserEntities();
+            Cache.set(CACHE_INTERNAL_ENTITIES, entities, 3600); // cache for an hour
```

Comment:
Will it change on an hourly basis? I think  24 hours should be a safe bet, since it's a better approx. of when a person is hired or leaves a company.